Fix github actions bug

mobiuscreek opened the pull request:
Self explanatory. Fixes the `add-path` issue caused by Github fixing a security vulnerability.

- https://www.zdnet.com/article/github-fixes-high-severity-security-flaw-spotted-by-google/
- https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/
- https://github.com/conda-incubator/setup-miniconda/issues/101#issue-744873629

metazool commented:
I think that we are not going to change this and it would be good to see it in main. 

Also if you want to track the model training (which is not yet covered by the tests) this changes to get MLFLOW_TRACKING_URI from the environment rather than hardcoding it

Alex-Hall-Data commented on corescore/models.py:
```diff
@@ -0,0 +1,102 @@
+            bs=self.batch_size, num_workers=0).normalize(imagenet_stats)  # hmm
+        return self.data
+
+    def acc_rock(self, input, target):
```
This actually makes sense now that I look at it again

Alex-Hall-Data reviewed (approved):
Tested on Ubuntu - cannot get fastai v1 to install at all on any windows machine. I don't think this is a huge issue though.

mobiuscreek committed:
Fix github actions bug
  - https://github.com/conda-incubator/setup-miniconda/issues/101#issue-744873629

mobiuscreek committed:
Run only pytest

mobiuscreek committed:
Add full build stage

mobiuscreek committed:
Use conda instead of pip

mobiuscreek committed:
Correct package names for conda

mobiuscreek committed:
Actually activate conda env

mobiuscreek committed:
Use same bash shell by default, more elegantly

mobiuscreek committed:
Downgrade to python 3.8 from 3.9

mobiuscreek committed:
Delete duplicate setting of python version

mobiuscreek commented:
I thought this would be simpler than changing the whole Github Actions workflow but now it's fine.

metazool commented:
Thank you for the fix. The conda setup is cleaner. I tried taking out the `defaults` to see what would happen (nothing!)

mobiuscreek commented:
Another reason was that pip tried and and [failed](https://github.com/BritishGeologicalSurvey/CoreScore/runs/1463209485#step:4:758)  to build a wheel for `opencv-python` after 40 minutes. Could probably disable pip's default behaviour by providing the `--no-binary`  argument but it looked cleaner in conda.
